guides/cookbook.md

```diff
@@ -169,6 +170,59 @@ int main(int argc, char **argv) {
 ```
 </div>
 
+<div class="tabpane C-sharp" markdown="1">
+Implementations of the IIonReader and IIonWriter interfaces are responsible for reading and writing
+Ion data in both text and binary forms.
+
+The following example shows how text Ion data can be read from a string:
+```c#
+using IonDotnet;
+using IonDotnet.Builders;
+using System;
+
+class ReadIonData
+{
+    static void Main(string[] args)
+    {
+        IIonReader reader = IonReaderBuilder.Build("{hello: \"world\"}");
+        reader.MoveNext();                           // position the reader at the first value, a struct
+        reader.StepIn();                             // step into the struct
+        reader.MoveNext();                           // position the reader at the first value in the struct
+        string fieldName = reader.CurrentFieldName;  // retrieve the current value's field name
+        string value = reader.StringValue();         // retrieve the current value's string value
+        reader.StepOut();                            // step out of the struct
+        Console.WriteLine(fieldName + " " + value);  // prints:  hello world
+    }
+}
+```
+
+In the above example, the text Ion `{hello: "world"}` was probably typed by a human using a text
+editor. The following example illustrates how it could have been generated using an IIonWriter:
+
+```c#
+using IonDotnet;
+using IonDotnet.Builders;
+using System;
+using System.IO;
+
+class WriteIonText
+{
+    static void Main(string[] args)
+    {
+        TextWriter tw = new StringWriter();
+        IIonWriter writer = IonTextWriterBuilder.Build(tw);
+        writer.StepIn(IonType.Struct);     // step into a struct
+        writer.SetFieldName("hello");      // set the field name for the next value to be written
+        writer.WriteString("world");       // write the next value
+        writer.StepOut();                  // step out of the struct
+        writer.Finish();
```

Comment:
This comment is applicable to all examples that use `IIonWriter`. Since `IIonWriter` implements `IDisposable`, its `Dispose()` method should be invoked (either directly or via `try/finally`) before it goes out of scope.

As an aside, should we make `IIonReader` implement `IDisposable`too, in case future implementations would need to use it to free resources?

Reply:
A [`using` declaration](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations) (or statement) is C#'s `try/finally` equivalent.  I've added `using` to all the `TextWriter` and `IIonWriter` declarations in this PR.

And yes, I'd say `IIonReader` should extend `IDisposable` too.  I've created https://github.com/amzn/ion-dotnet/issues/70 to track.